### PR TITLE
Fix scratch and multilevel tests

### PR DESCRIPTION
**Pull request by drwells**
I am working on a new IBFE example and I noticed that the scratch multilevel tests aren't right - they don't actually use multiple levels. It turns out the other tests I added don't either. Fortunately you can see that they don't make a big difference in the values we compute (which implies that the multilevel stuff works).

Another fix: we were not regridding all scratch hierarchy levels which lead to some poor load balancing. This is also fixed.

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

File: src/IB/IBFEMethod.cpp [hunk 1]
```diff
@@ -1506,12 +1506,11 @@ void IBFEMethod::endDataRedistribution(Pointer<PatchHierarchy<NDIM> > /*hierarch
...
             // any sort of tagging based on the current time I think we can
             // just put in a bogus (nonnegative) value.
-            int finest_level = d_scratch_hierarchy->getFinestLevelNumber();
-            if (finest_level == 0)
+            if (getCoarsestPatchLevelNumber() == 0)
                 d_scratch_gridding_algorithm->makeCoarsestLevel(d_scratch_hierarchy, 0.0 /*d_current_time*/);
             else
                 d_scratch_gridding_algorithm->regridAllFinerLevels(
-                    d_scratch_hierarchy, finest_level - 1, 0.0 /*d_current_time*/, tag_buffer);
+                    d_scratch_hierarchy, getCoarsestPatchLevelNumber() - 1, 0.0 /*d_current_time*/, tag_buffer);
```
Comment (boyceg) [inline src/IB/IBFEMethod.cpp:1513]:
```suggestion
                    d_scratch_hierarchy, std::max(getCoarsestPatchLevelNumber() - 1, 0), 0.0 /*d_current_time*/, tag_buffer);
```
?

**Review by boyceg — APPROVED** [review]
Looks good.

Only comment is that I am not sure if `regridAllFinerLevels()` needs to have a valid level number. It seems like the level number could be invalid if `getCoarsestPatchLevelNumber()` is `0`.

File: src/IB/IBFEMethod.cpp (hunk 1 above)
Comment (drwells) [reply src/IB/IBFEMethod.cpp:1513]: See like 1509 :)

Comment (boyceg) [reply src/IB/IBFEMethod.cpp:1513]: Oh right.

Comment (boyceg) [reply src/IB/IBFEMethod.cpp:1513]: Should there be a subsequent call to `regridAllFinerLevels()` if `getCoarsestPatchLevelNumber()` returns `0`?

Comment (drwells) [reply src/IB/IBFEMethod.cpp:1513]: I think you're right. Let me take a look.

**Comment by drwells** [comment]
I fixed the problem with the coarsest level - the scratch hierarchy now works with all supported possibilities.

**Comment by drwells** [comment]
Lets merge this before we go on to #1137.

**Review by boyceg — APPROVED** [review]

**Pull request merged**